src/App/Router.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Friendica\App;
+
+
+use FastRoute\DataGenerator\GroupCountBased;
+use FastRoute\Dispatcher;
+use FastRoute\RouteCollector;
+use FastRoute\RouteParser\Std;
+use Friendica\Module\Itemsource;
+
+class Router
+{
+	/** @var RouteCollector */
+	public $routeCollector;
+
+	public function __construct(RouteCollector $routeCollector = null)

```

Comment:
Is this parameter really optional? Creating an instance inside the constructor seems like hiding a failure.

Reply:
It's because the `RouteCollector` object needs two more class instances (a `RouteParser` and a `DataGenerator`) that only make sense in the context of the router and are more settings with sensible defaults.

If we want to change those defaults, we can supply an already existing `RouteCollector` object with different sub-behaviors, but otherwise I don't mind instantiating the `RouteCollector` in the Router constructor.

Reply:
I see. This could be usefull for testing of course :-)

Reply:
That's exactly why I kept the constructor parameter!